systemtest/src/test/java/io/strimzi/systemtest/specific/SpecificST.java

```diff
@@ -59,15 +56,25 @@ void testRackAware() throws Exception {
                 .endKafka()
             .endSpec().done();
 
+        Affinity kafkaPodSpecAffinity = kubeClient().getStatefulSet(KafkaResources.kafkaStatefulSetName(CLUSTER_NAME)).getSpec().getTemplate().getSpec().getAffinity();
+        NodeSelectorRequirement kafkaPodNodeSelectorRequirement = kafkaPodSpecAffinity.getNodeAffinity()
+                .getRequiredDuringSchedulingIgnoredDuringExecution().getNodeSelectorTerms().get(0).getMatchExpressions().get(0);
+
+        assertThat(kafkaPodNodeSelectorRequirement.getKey(), is(rackKey));
+        assertThat(kafkaPodNodeSelectorRequirement.getOperator(), is("Exists"));
+
+        PodAffinityTerm kafkaPodAffinityTerm = kafkaPodSpecAffinity.getPodAntiAffinity().getPreferredDuringSchedulingIgnoredDuringExecution().get(0).getPodAffinityTerm();
+
+        assertThat(kafkaPodAffinityTerm.getTopologyKey(), is(rackKey));
+        assertThat(kafkaPodAffinityTerm.getLabelSelector().getMatchLabels(), hasEntry("strimzi.io/cluster", CLUSTER_NAME));
+        assertThat(kafkaPodAffinityTerm.getLabelSelector().getMatchLabels(), hasEntry("strimzi.io/name", KafkaResources.kafkaStatefulSetName(CLUSTER_NAME)));
+
         String rackId = cmdKubeClient().execInPod(KafkaResources.kafkaPodName(CLUSTER_NAME, 0), "/bin/bash", "-c", "cat /opt/kafka/init/rack.id").out();
         assertThat(rackId.trim(), is("zone"));
 
-        String brokerRack = cmdKubeClient().execInPod(KafkaResources.kafkaPodName(CLUSTER_NAME, 0), "/bin/bash", "-c", "cat /tmp/strimzi.properties | grep broker.rack").out();
-        assertThat(brokerRack.contains("broker.rack=zone"), is(true));
+        String brokerRack = cmdKubeClient().execInPod(KafkaResources.kafkaPodName(CLUSTER_NAME, 0), "/bin/bash", "-c", "cat /tmp/strimzi.properties | grep rack.id").out();
```

Comment:
I think you found a bug here ... so you should not change the test accept the bug ;-). I opened #2550 to fix it.

Reply:
hehe, I will change it back then :)